amt_test.go

```diff
@@ -44,14 +45,37 @@ func TestMain(m *testing.M) {
 
 type mockBlocks struct {
 	data               map[cid.Cid]block.Block
+	dataMu             sync.Mutex
```

Comment:
required for accessing the blockstore in parallel (as is done by ParallelDiff), the race detector throws a fit otherwise.